src/NzbDrone.Core/Update/InstallUpdateService.cs

```diff
@@ -108,13 +119,35 @@ private void InstallUpdate(UpdatePackage updatePackage)
                                             updateSandboxFolder);
 
                 _logger.Info("Starting update client {0}", _appFolderInfo.GetUpdateClientExePath());
-                _logger.ProgressInfo("NzbDrone will restart shortly.");
+                _logger.ProgressInfo("Sonarr will restart shortly.");
 
                 _processProvider.Start(_appFolderInfo.GetUpdateClientExePath(), GetUpdaterArgs(updateSandboxFolder));
+
+                return;
             }
-            catch (Exception ex)
+            catch (UpdateFailedException ex)
             {
                 _logger.ErrorException("Update process failed", ex);
+                throw;
+            }
+        }
+
+        private void SwitchUpdatePackageBranch(UpdatePackage package)
```

Comment:
The method name implies, switch is happening, we should either get rid of the if in this method, or call it `EnsureValidBranch` or something similar.